File: day1/one.hs
main = do
    input <- readFile "input"
    print $ start $ map read $ lines input


start :: [Int] -> Int
start (fst:snd:trd:tail) = process (fst + snd + trd) $ snd:trd:tail
start _ = 0

process :: Int -> [Int] -> Int
process last (fst:snd:trd:tail) = (if curr > last then 1 else 0) + process curr (snd:trd:tail)
    where
        curr = fst + snd + trd
process _ _ = 0

File: day1/one.py
count = 0
with open('input', 'r') as input:
    fst, snd, trd = (int(input.readline()) for _ in range(3))
    last = fst + snd + trd
    for line in input.readlines():
        fst, snd, trd = (snd, trd, int(line))
        curr = fst + snd + trd
        if curr > last:
            count += 1
        last = curr
print(count)